mlflow/store/artifact/databricks_artifact_repo.py

```diff
@@ -231,6 +238,38 @@ def _azure_upload_file(self, credentials, local_file, artifact_path):
         except Exception as err:
             raise MlflowException(err)
 
+    def _azure_adls_gen2_upload_file(self, credentials, local_file, artifact_path):
+        """
+        Uploads a file to a given Azure storage location using the ADLS gen2 API.
+        """
+        try:
+            headers = self._extract_headers_from_credentials(credentials.headers)
+
+            # try to create the file
+            try:
+                put_adls_file_creation(credentials.signed_uri, headers=headers)
+            except requests.HTTPError as e:
+                raise e
+
+            # next try to patch the file
+            offset = 0
+            for chunk in yield_file_in_chunks(local_file, _AZURE_MAX_BLOCK_CHUNK_SIZE):
+                try:
+                    patch_adls_file_upload(
+                        credentials.signed_uri, chunk, str(offset), headers=headers
```

Comment:
Nit: I think `patch_adls_file_upload` and `patch_adls_flush` should do the string conversion of the offset, not the caller

Reply:
Fixed.